client/state/sites/plans/actions.js

```diff
@@ -127,7 +127,7 @@ export function fetchSitePlansCompleted( siteId, data ) {
 	let plansData = omit( data, '_headers' );
 	const { product_id } = personalPlan;
 
-	if ( ! has( plansData, product_id ) ) {
+	if ( ! isEmpty( plansData ) && ! plansData.hasOwnProperty( product_id ) ) {
```

Comment:
It was intentional because some test send empty data to the action creator, and expect an empty array back.